### PR TITLE
Issue #3284885 by vnech: Fix comments field place in entity form creation page

### PR DESCRIPTION
**Pull request by nechai**
## Problem
Sometimes "Comments Settings" field is placed out of the "Settings" group field on entity create page.

Screenshot 2022-06-09 at 18 02 15

## Solution
Hardcode comments field in "settings" field group.

## Issue tracker
- https://www.drupal.org/project/social/issues/3284885
- https://getopensocial.atlassian.net/browse/YANG-7700

## Theme issue tracker
N/A

## How to test
- [ ] Login as an admin
- [ ] Create a new content type - `Dummy`
- [ ] Implement `hook_social_content_type_alter()` and `hook_social_core_compatible_content_forms_alter()` for the node type
- [ ] Enable comments in this content type
- [ ] Visit add/edit pages for the content type

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes
*Fix comments field place in entity form creation page.*

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2984-uiqjokvkb5opjycj8sqh1feethu5agkv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62a20cd16eb9e7df1aa1e820

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2984-uiqjokvkb5opjycj8sqh1feethu5agkv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62a20cd16eb9e7df1aa1e820

**Comment by ribel** [comment]
@rollki @nechai please add correct milestone here.

**Comment by nechai** [comment]
Hi @Ressinel! The `comments` can be added to the entites which aren't a part of the distro. The current fix unify `comments` field visibility for all entities.

**Comment by Ressinel** [comment]
@nechai In this case, could you not update the HTT section with appropriate steps? Thanks!

File: modules/social_features/social_core/social_core.module [hunk 1]
```diff
@@ -990,6 +991,25 @@ function social_core_form_node_form_alter(&$form, FormStateInterface $form_state
   }
 }
 
+/**
+ * Implements hook_form_alter().
+ */
+function social_core_form_alter(array &$form, FormStateInterface $form_state, string $form_id): void {
+  if (method_exists($form_state->getFormObject(), 'getEntity')) {
```
Comment (Ressinel) [inline modules/social_features/social_core/social_core.module:998]: @nechai This code will move the comments field to the Settings group only if `$form['advanced']` does not exist. Please consider this. And add a comment with clarification, please.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2984-uiqjokvkb5opjycj8sqh1feethu5agkv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62a20cd16eb9e7df1aa1e820

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2984-uiqjokvkb5opjycj8sqh1feethu5agkv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62a20cd16eb9e7df1aa1e820

**Review by Ressinel — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2984-uiqjokvkb5opjycj8sqh1feethu5agkv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62a20cd16eb9e7df1aa1e820

**Pull request merged**